software/follow_ball_depth.py

```diff
@@ -0,0 +1,300 @@
+
+        def run(self):
+            print("Starting " + self.name)
+            try:
+                self.rf_command.connect()
+            except:
+                print("can not connect to server")
+                self.interupt = True
+            
+            while True:
+
+                msg = json.loads(self.rf_command.listen())
+                print(msg)
+                try :
+                    index_robot = msg.get("targets").index("robodendron") 
+                except:
+                    break
+                if msg.get("targets")[index_robot] == "robodendron":
+                    if msg.get("signal") == "stop":
+                        self.interupt = True
+                        robot.state = states.find_ball
+                        print(msg.get("signal"))
+
+                    else:
+                        robot.target_basket = msg.get("baskets")[index_robot]
+                        if robot.target_basket == 'blue':
+                            robot.oponent_basket = 'magenta'
+                        else:
+                            robot.oponent_basket = "blue"
+
+                        print(msg.get("signal"),"bakset target : ",robot.target_basket)
+                        self.interupt = False
+
+
+    class align_control(threading.Thread):
+    
+        def __init__(self, threadID, name,processor):
+            
+            threading.Thread.__init__(self)
+            self.threadID = threadID
+            self.name = name
+            self.processor = processor
+        
+        def run(self):
+            print("Starting " + self.name)
+
+        def speed_thrower(self,dist):
+            speed = min(int(0.240*dist + 415),1900)
+            return speed
+        
+        
+        def throw(self, depth_frame):
+            distance = depth_frame[self.basket_.y, self.basket_.x]
+            print("------------------------------------")
+            print(distance)
+            print("------------------------------------")
+            time.sleep(0.2)
+            robot.motion_sim3.thrower_control(600)
+            time.sleep(0.2)
+            print('dist basket',self.basket_.y)
+            thr_speed = self.speed_thrower(distance)
+            print('speed',thr_speed)
+            robot.motion_sim3.move(6,-0.15,0,thr_speed) 
+            time.sleep(1)
+            robot.motion_sim3.thrower_control(300)
+
+
+        def align_on(self):
+            processedData = self.processor.process_frame(aligned_depth=False)
+            if robot.state == states.align_basket:
+                largest = max(processedData.balls , key = lambda ball: ball.size, default=None)
+                if robot.target_basket == 'blue':
+                    self.basket_ = processedData.basket_b
+                else:
+                    self.basket_ = processedData.basket_m
+                
+                if largest:
+        
+                    robot.speedz = (processedData.debug_frame.shape[1]/2) - largest.x 
```

Comment:
Kinda in-intuitive to get frame size from the debug frame. If you need to use the sizes, I suggest putting this data into the processedData